src/batch.js

```diff
@@ -1,28 +1,221 @@
 'use strict'
 
+const router = require('lambda-router/src/router')
+const { expectation } = require('sinon')
+
 const { HttpError } = require('./httpError')
 
 const batchDefaultConfig = {
-  maxBatchSize: 20
+  maxBatchSize: 20,
+  handler: undefined
 }
 
 module.exports = {
   batchHandler
 }
 
-async function batchHandler({ routes, config, extractPathParameters }, event, context) {
+async function batchHandler({ route, config }, event, context) {
   config = { ...batchDefaultConfig, ...config }
+  const body = event.body
 
   // Validate
-  // validate json body has been parsed
-  // validate body.requests has array of requests
-  // validate each body.requests[]
+  validateBatchRequest(body, config.maxBatchSize)
 
   // build graph of requests
+  const orderedRequests = tSort(body.requests)
+  const { asyncRequests, syncRequests } = splitAsyncRequests(orderedRequests)
 
   // execute requests
+  const results = await Promise.all([
+    (async () => {
+      const syncResults = []
+      for (let req of syncRequests) {
+        syncResults.push(await executeRequest(config, route, event, req))
+      }
+      return syncResults
+    })(),
+    ...asyncRequests.map(executeRequest.bind(null, config, route, event))
+  ])
```

Comment:
Attempting to make sure the async requests are executed simultaneously with the sync requests, but also make sure sync request are executed in order between themselves.

Reply:
Promise.all() opens up the potential for unexpected behavior if any of the routes throw uncaught exceptions.

We could use Promise.allSettled(), but it would require making this library require node.js v12.9

Reply:
Using `Promise.all` with a `catch` wrapped onto each promise is a close-enough way to mimic `allSettled`